test/integration/reasoner/ResolutionTest.java

```diff
@@ -0,0 +1,451 @@
+        String atomic2 = "$p1 isa person; $p1 has name \"Alice\"";
+        String atomic3 = "$p1 isa person; $p1 has age 24;";
+        String atomic4 = "$p1 isa person; $p1 has name \"Bob\";";
+        String atomic5 = "$p1 isa person; $p1 has age 72;";
+        try (RocksSession session = schemaSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().define(Graql.parseQuery(
+                        "define person sub entity, owns age, owns name, plays twins:twin1, plays twins:twin2;" +
+                                "age sub attribute, value long;" +
+                                "name sub attribute, value string;" +
+                                "twins sub relation, relates twin1, relates twin2;"));
+                transaction.commit();
+            }
+        }
+        try (RocksSession session = dataSession()) {
+            try (RocksTransaction transaction = singleThreadElgTransaction(session)) {
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic1));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic2));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic2));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic2));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic3));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic3));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic3));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic4));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic4));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic4));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic5));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic5));
+                transaction.query().insert(Graql.parseQuery("insert " + atomic5));
+                transaction.commit();
+            }
+        }
+        long atomic1TraversalAnswerCount = 3L;
+        long atomic2TraversalAnswerCount = 3L;
+        long atomic3TraversalAnswerCount = 3L;
+        long atomic4TraversalAnswerCount = 3L;
+        long atomic5TraversalAnswerCount = 3L;
+        long conjunctionTraversalAnswerCount = 0L;
+        long answerCount = conjunctionTraversalAnswerCount + (atomic5TraversalAnswerCount * atomic4TraversalAnswerCount * atomic3TraversalAnswerCount * atomic2TraversalAnswerCount * atomic1TraversalAnswerCount);
+        Conjunction conjunctionPattern = parseConjunction("{ " + atomic5 + " " + atomic4 + " " + atomic3 + " " + atomic2 + " " + atomic1 + " }");
```

Comment:
are you intentionally keeping multiple `isa` on `$p1` after merging? I think it's fine as they get deduplicated, just want to check